tensorflow/conv1d.py

```diff
@@ -0,0 +1,236 @@
+    filter_input_channels: int,
+    filter_output_channels: int,
+):
+    g = tf.Graph()
+    with tf.device("/%s:0" % (_ARGS_DEVICE)) and g.as_default():
+        if _ARGS_DATA_FORMAT == "NWC":
+            init_tensor = tf.Variable(
+                tf.ones(
+                    [
+                        batch,
+                        tensor_input_width,
+                        tensor_input_channels,
+                    ]
+                ),
+                dtype=_ARGS_DTYPE,
+            )
+        elif _ARGS_DATA_FORMAT == "NCW":
+            init_tensor = tf.Variable(
+                tf.ones(
+                    [
+                        batch,
+                        tensor_input_channels,
+                        tensor_input_width,
+                    ]
+                ),
+                dtype=_ARGS_DTYPE,
+            )
+        else:
+            raise ValueError("Unknown data_format: " + str(_ARGS_DATA_FORMAT))
+     
+        init_filter = tf.Variable(
+            tf.ones(
+                [
+                    filter_width,
+                    filter_input_channels,
+                    filter_output_channels,
+                ]
+            ),
+            dtype=_ARGS_DTYPE,
+        )
+        convolution = tf.nn.conv1d(
+            init_tensor,
+            filters=init_filter,
+            stride=_ARGS_STRIDE,
+            padding=_ARGS_PADDING,
+            data_format=_ARGS_DATA_FORMAT,
+        )
+
+    times = []
+    config = tf.ConfigProto()
+    with tf.Session(graph=g, config=config) as sess:
+        sess.run(tf.global_variables_initializer())
+        # warmup
+        sess.run(convolution.op)
+
+        for i in range(_ARGS_REPS):
+            start = time.time()
+            sess.run(convolution.op)
+            times.append(time.time() - start)
+
+    times_ms = 1000 * np.array(times)  # in seconds, convert to ms
+    elapsed_ms = np.median(times_ms)
+    # Formula:
+    #  batch_size * x_dim * kernel_x_dim 
+    #  * input_depth * output_depth * 2 / (x_stride)
+    ops = (
+        batch
+        * tensor_input_width
+        * filter_width
+        * tensor_input_channels
+        * filter_output_channels
+        * 2
+    ) / (_ARGS_STRIDE)
+    print(ops)
+    rate = ops / elapsed_ms / 10 ** 6  # in GFLOPS. (/ milli / 10**6) == (/ 10 ** 9)
+    print('conv took:   \t%.4f ms,\t %.2f GFLOPS' % (elapsed_ms, rate), file=sys.stderr)
+
+    return rate, elapsed_ms
+
+
+def main():
+    np.set_printoptions(suppress=True)
+    print("# Version: %s, path: %s" % (tf.__version__, tf.__path__), file=sys.stderr)
+
+    rate, elapsed = bench(
+        batch=_ARGS_BATCH,
+        tensor_input_width=_ARGS_INPUT_WIDTH,
+        tensor_input_channels=_ARGS_T_INPUT_CHANNELS,
+        filter_width=_ARGS_FILTER_WIDTH,
+        filter_input_channels=_ARGS_F_INPUT_CHANNELS,
+        filter_output_channels=_ARGS_OUTPUT_CHANNELS
+    )
+
+    result = {
+        "framework": "tensorflow",
+        "name": "conv1D",
```

Comment:
`conv1d`?

Reply:
also in conv2d.py -> `conv2d`?

Reply:
Yes, for natural language processing applications, you can use also convolution. In that case you require conv1d to extract features from a sentence, which is 1d typically.

Reply:
I was referring to uppercase D in the name - do we want to keep it uppercase?

Reply:
It depends on the dimension of convolution you want to apply in general. There is also conv3d.

Reply:
Updated with lower case, thanks @fridex :beers: